### PR TITLE
Setup game team

### PR DESCRIPTION
**Pull request by perryr16**
Game_team class created, Stat_tracker created, team_test

### CODE REVIEW

File: lib/game_team.rb [hunk 1]
```diff
@@ -0,0 +1,48 @@
+class GameTeam
+#
+  @@all = nil
+
+  def self.all
+    @@all
+  end
+
```
Comment (perryr16) [inline lib/game_team.rb:8]: this method could be pulled out with inheritence

File: lib/game_team.rb [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    @game_id = details[:game_id].to_i
+    @team_id = details[:team_id].to_i
+    @hoa = details[:hoa]
+    @result = details[:result]
+    @settled_in = details[:settled_in]
+    @head_coach = details[:head_coach]
+    @goals = details[:goals].to_i
+    @shots = details[:shots].to_i
+    @tackles = details[:tackles].to_i
+    @pim = details[:pim].to_i
```
Comment (perryr16) [inline lib/game_team.rb:40]: note that the headers are camel case, then when pulled in they are downcased. Ruby convention dictates snake case

File: lib/stat_tracker.rb [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    games_path = file_paths[:games]
+    teams_path = file_paths[:teams]
+
+    self.new(game_teams_path, games_path, teams_path)
+
+  end
+
+  attr_reader :game_teams_path,
+              :games_path,
+              :teams_path,
```
Comment (perryr16) [inline lib/stat_tracker.rb:23]: Aurora said not to run the .from_csv every time we want to access the collection. This runs 3 collection methods to create the collection instance variables

File: runner.rb [hunk 4]
```diff
@@ -0,0 +1,17 @@
+# runner.rb
```
Comment (perryr16) [inline runner.rb:1]: straight from iteration 1

File: test/fixtures/short_game_teams.csv [hunk 5]
```diff
@@ -0,0 +1,4 @@
+game_id,team_id,HoA,result,settled_in,head_coach,goals,shots,tackles,pim,powerPlayOpportunities,powerPlayGoals,faceOffWinPercentage,giveaways,takeaways
```
Comment (perryr16) [inline test/fixtures/short_game_teams.csv:1]: fixtures files for testing

File: test/game_team_test.rb [hunk 6]
```diff
@@ -0,0 +1,81 @@
+require 'simplecov'
+SimpleCov.start
```
Comment (perryr16) [inline test/game_team_test.rb:2]: not sure why my coverage istn higher

File: test/game_team_test.rb [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+require 'minitest/autorun'
+require 'minitest/pride'
+require 'pry'
+require './lib/stat_tracker'
+require './lib/game_team'
+#
+class GameTeamTest < Minitest::Test
+
+  def setup
```
Comment (perryr16) [inline test/game_team_test.rb:11]: setup a hardcoded game-team, and a dynamic game-team

File: test/game_team_test.rb [hunk 8]
```diff
@@ -0,0 +1,81 @@
...
+    assert_instance_of GameTeam, @base_game_team
+    assert_instance_of GameTeam, @game_teams.first
+  end
+
+  def test_it_returns_list_of_game_teams
+    assert_instance_of Array, GameTeam.all
+    assert_equal 3, GameTeam.all.length
+    assert_instance_of GameTeam, GameTeam.all.first
+  end
+
```
Comment (perryr16) [inline test/game_team_test.rb:45]: test attribute return for hard-coded and dynamic

File: test/stat_tracker_test.rb [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+SimpleCov.start
+require 'minitest/autorun'
+require 'minitest/pride'
+require 'pry'
+require './lib/stat_tracker'
+require './lib/game'
+require './lib/team'
+require './lib/game_team'
+#
+
```
Comment (perryr16) [inline test/stat_tracker_test.rb:11]: tests will fail until other classes are created

File: lib/game_team.rb (hunk 1 above)
Comment (stellakunzang) [reply lib/game_team.rb:8]: good to note!

File: lib/stat_tracker.rb (hunk 3 above)
Comment (stellakunzang) [reply lib/stat_tracker.rb:23]: thank you for solving this problem. I hadn't yet figured it out!

Comment (foleymichelle9) [reply lib/stat_tracker.rb:23]: wow! this is great it will definitely help us out in the long run!

File: test/fixtures/short_game_teams.csv (hunk 5 above)
Comment (stellakunzang) [reply test/fixtures/short_game_teams.csv:1]: I did not create a fixture file for testing game class, but called full CSV. It takes a long time and I will do this now!

File: test/game_team_test.rb (hunk 6 above)
Comment (stellakunzang) [reply test/game_team_test.rb:2]: I have basically the same thing (minus using fixture file) and my coverage is 100% so also not sure

**Review by stellakunzang — APPROVED** [review]
I approve merging these changes. It looks like a perfect starting point to get our project rolling.

File: test/game_team_test.rb (hunk 6 above)
Comment (foleymichelle9) [reply test/game_team_test.rb:2]: We'll keep an eye on it as we go but my theory is that not everything has a test written for it? It's probably something small :)

**Comment by foleymichelle9** [comment]
Agreed everything looks great!

**Comment by Rostammahabadi** [comment]
Everything looks good to go! Great job on the descriptions of the comments

**Pull request merged**